backend/substrapp/tasks/tasks_save_image.py

```diff
@@ -89,9 +108,16 @@ def save_image_task(task: SaveImageTask, function_serialized: str, channel_name:
 
         logger.info("Start saving the serialized image")
         # save it
-        FunctionImage.objects.create(
+        image = FunctionImage.objects.create(
             function_id=function.key, file=File(file=storage_path.open(mode="rb"), name="image.zip")
         )
+        # update APIFunction image-related fields
+        api_function = ApiFunction.objects.get(key=function.key)
+        # TODO get full url cf https://github.com/Substra/substra-backend/backend/api/serializers/function.py#L66
+        api_function.image_address = settings.DEFAULT_DOMAIN + reverse("api:function-image", args=[function.key])
```

Comment:
As we use this function twice in this file, is it worth creating a function containing this logic?

Reply:
This still feels like a quite hacky way of obtaining the URL, I will open a follow-up issue.